Plugins/Detray/include/Acts/Plugins/Detray/DetrayConverter.hpp

```diff
@@ -0,0 +1,146 @@
+                                   const Surface& surface, bool portal = false);
+/// Conversion method for Portal object to detray::portal payloads
+///
+/// @param gctx the geometry context
+/// @param portal the portal to be converted
+/// @param ip the portal index
+/// @param volume the volume to which the portal belongs
+/// @param orientedSurfaces the oriented surfaces of the portal
+/// @param detectorVolumes the detector volumes for the link lookup
+///
+/// @note due to portal splitting this can add up in N portals for one initial one
+///
+/// @brief convert the acts portal to detray surface payload and populate the payload
+std::vector<io::surface_payload> convertPortal(
+    const GeometryContext& gctx, const Experimental::Portal& portal,
+    std::size_t ip, const Experimental::DetectorVolume& volume,
+    const std::vector<OrientedSurface>& orientedSurfaces,
+    const std::vector<const Experimental::DetectorVolume*>& detectorVolumes);
+
+/// Conversion method for volume objects to detray::volume payloads
+///
+/// @param gctx the geometry context
+/// @param volume the volume to be converted
+/// @param detectorVolumes the detector volumes for the link lookup
+///
+/// @return the volume_payload for portals and volumes by @param volume acts object
+io::volume_payload convertVolume(
+    const GeometryContext& gctx, const Experimental::DetectorVolume& volume,
+    const std::vector<const Experimental::DetectorVolume*>& detectorVolumes);
+
+/// Conversion method for (common) header payload
+///
+/// @param detector is the detector to be converted
+///
+/// @return a geometry header payload
+io::geo_header_payload convertHead(
+    const Acts::Experimental::Detector& detector);
+
+/// Convert an Acts::Experimental::Detector to a detray::detector object
+///
+/// @param gctx the geometry context
+/// @param detector the detector to be converted
+/// @param mr the memory resource to be used
+///
+/// @returns a detector of requested return type
+template <typename detector_t = DetrayDetector>
+std::tuple<detector_t, vecmem::memory_resource&> convertDetector(
+    const Acts::GeometryContext& gctx,
+    const Acts::Experimental::Detector& detector, vecmem::memory_resource& mr) {
+  detray::io::detector_payload detecorPayload;
+  for (const auto volume : detector.volumes()) {
+    detecorPayload.volumes.push_back(
+        convertVolume(*volume, detector.volumes(), gctx));
+  }
+  typename detector_t::name_map names = {{0u, detector.name()}};
+
+  // build detector
+  detector_builder<default_metadata> detectorBuilder{};
+  detray::io::geometry_reader::convert<detector_t>(detectorBuilder, names,
+                                                   detectorPayload);
+  detector_t detrayDetector(detectorBuilder.build(mr));
+
+  // checks and print
+  detray::detail::check_consistency(detrayDetector);
+  converterPrint(detrayDetector, names);
+
+  return {std::move(det_tuple), mr};
```

Comment:
What is ```det_tuple```?

Reply:
It used to be detector tuple. Not anymore :/